app/assets/javascripts/availability.js.coffee

```diff
@@ -0,0 +1,29 @@
+jQuery ->
+  window.availability_updater = new AvailabilityUpdater
+class AvailabilityUpdater
+  constructor: ->
+    this.request_availability()
+  availability_url: "http://bibdata.princeton.edu/availability"
+  request_availability: ->
+    ids = this.record_ids().toArray()
+    params = $.param({ids: ids})
+    url = "#{@availability_url}?#{params}"
+    $.getJSON(url, this.process_records)
+  process_records: (records) =>
+    for record_id, availability_info of records
+      this.apply_record(record_id, availability_info)
+  apply_record: (record_id, availability_info) ->
+    return if availability_info.more_holdings == true
+    for location_code, status of availability_info
+      element = $("*[data-record-id='#{record_id}'][data-loc-code='#{location_code}']")
+      continue unless element.length > 0
+      availability_element = element.children(".availability-icon")
+      availability_element.addClass("glyphicon")
+      if status == "Not Charged"
```

Comment:
It turns out that available/unavailable is more involved than "Not charged"/else. I'd be ok with merging as is if we open up a new ticket about how to classify based on some of the other item statuses mentioned in the [issue](https://github.com/pulibrary/orangelight/issues/107#issuecomment-147882725). I'm thinking that logic may ultimately end up in bibdata.